captum/influence/_core/tracincp_fast_rand_proj.py

```diff
@@ -663,6 +664,7 @@ def _self_influence(self):
         )
         raise NotImplementedError
 
+    @log_usage()
```

Comment:
I think that since this method is not even abstract I'm skeptical that the logging will work for child classes. @vivekmig, @aobo-y  what do you think ?

Reply:
I read through the code and actually feel it should word. Abstract or not does not matter. The `log_usage` can decorate any functions. But of course, juts based on my brain parsing. @99warriors  have you tested the logging internally?

Reply:
yeah, it looks like we have the definitions only in the abstract class. I kept thinking about the attributions, where we have overrides. It probably should be fine but it would be good to test.